internal/xpkg/dep/cache/entry.go

```diff
@@ -246,23 +246,12 @@ func (e *entry) writeObjects(objs []runtime.Object) (*flushstats, error) { // no
 			continue
 		}
 
-		f, err := e.fs.Create(filepath.Join(e.location(), fmt.Sprintf(crdNameFmt, name)))
-		if err != nil {
-			return stats, err
-		}
-		defer f.Close() // nolint:errcheck
-
-		fb, err := f.Write(yb)
-		if err != nil {
-			return stats, err
-		}
-
-		if fb == 0 {
-			return stats, errors.New(errFailedToCreateCRD)
+		if err := os.WriteFile(filepath.Join(e.location(), fmt.Sprintf(crdNameFmt, name)), yb, 0o600); err != nil {
```

Comment:
I think `os.Writefile` is going to side step the use of `afero.Fs` here. We can fix that up by using afero's API xref https://github.com/spf13/afero#using-aferos-utility-functions. So something like this:

```suggestion
		if err := afero.WriteFile(e.fs, filepath.Join(e.location(), fmt.Sprintf(crdNameFmt, name)), yb, 0o600); err != nil {
```
may work. Here's another example of doing that in this repo https://github.com/upbound/up/blob/main/internal/xpkg/build_test.go#L171.

Reply:
Hi @tnthornton,
Thanks, done.